Dto impl

shrralis opened the pull request:
The most important changes is on commit changelog #35:

##    [GLOBAL]:
     - We're using Java8 now and Spring 4 instead of Java9 and Spring 5.
     - Password recovery added. [need to finish and test with front-end]

##    [Front-end]:
     - Added password recovery form.
     - Some code cleans.
     - Added showing some errors (if we retrieved them) from back-end.

##    [Back-end]:
     - We have mappers for converting entities into DTO and back.
     - We can send E-Mails now.
     - Added `@Transactional` to all of the services.
     - Now repositories are not as `@Resource`, they're `@Autowired`.
     - Spring Security configuration rearranged a little.
     - Removed unnecessary `GeneralRestController`.
     - Now we have Apache Commons dependency.

##    [Database]:
     - New table for password recovery.

shrralis committed:
# Shrralis commit changelog #35:

    [GLOBAL]:
     - We're using Java8 now and Spring 4 instead of Java9 and Spring 5.
     - Password recovery added. [need to finish and test with front-end]

    [Front-end]:
     - Added password recovery form.
     - Some code cleans.
     - Added showing some errors (if we retrieved them) from back-end.

    [Back-end]:
     - We have mappers for converting entities into DTO and back.
     - We can send E-Mails now.
     - Added `@Transactional` to all of the services.
     - Now repositories are not as `@Resource`, they're `@Autowired`.
     - Spring Security configuration rearranged a little.
     - Removed unnecessary `GeneralRestController`.
     - Now we have Apache Commons dependency.

    [Database]:
     - New table for password recovery.

shrralis committed:
Merge branch 'dev' into dto_impl

# Conflicts:
#	front-end/src/router/index.js
#	src/main/java/com/shrralis/ssdemo1/controller/IssueRestController.java
#	src/main/java/com/shrralis/ssdemo1/entity/Issue.java
#	src/main/java/com/shrralis/ssdemo1/service/MapMarkersServiceImpl.java
#	src/main/java/com/shrralis/ssdemo1/service/interfaces/IIssueService.java
#	src/main/java/com/shrralis/ssdemo1/service/interfaces/IMapMarkersService.java

shrralis committed:
# Shrralis commit changelog #36:

    [GLOBAL]:
     Merged with last `dev` branch.

    [Front-end]:
     `null`

    [Back-end]:
     `null`

    [Database]:
	 `null`

yurakiziuk commented on pom.xml:
```diff
@@ -307,12 +350,35 @@
-        <dependency>
-	        <groupId>com.sun.mail</groupId>
-	        <artifactId>javax.mail</artifactId>
+	    <!--<dependency>-->
```
No comments. Just remove it

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/configuration/AppConfig.java:
```diff
@@ -29,9 +41,18 @@
+	public static Boolean DEBUG;
+
+	@Resource
+	private Environment env;
```
Use @Value

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/configuration/AppConfig.java:
```diff
@@ -40,4 +61,44 @@ public void addCorsMappings(CorsRegistry registry) {
+	@Override
+	public void extendMessageConverters(List<HttpMessageConverter<?>> converters) {
+		for (HttpMessageConverter converter : converters) {
+			if (converter instanceof MappingJackson2HttpMessageConverter) {
```
I believe it should be another way to configure jackson mapper. Have a fun =)

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/configuration/AppConfig.java:
```diff
@@ -29,9 +41,18 @@
+	private static final String EMAIL_PASS = "email.password";
+
+	@Value("${environment.debug}")
+	public static Boolean DEBUG;
```
Why static ?

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/configuration/SecurityConfigAdapter.java:
```diff
@@ -0,0 +1,79 @@
+				.cors()
+				.and()
+				.authorizeRequests()
+//					next line should allow passing all unauthorized requests
```
No comments

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/controller/AuthRestController.java:
```diff
@@ -12,96 +12,64 @@
-        final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+	@GetMapping(value = "/login", params = "error")
+	public JsonResponse showErrorOnLogin() {
+//		TODO: finish this
```
Just do it

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/controller/AuthRestController.java:
```diff
@@ -12,96 +12,64 @@
-    }
+	@GetMapping({ "/login", "/logout" })
+	public JsonResponse getCurrSession() {
+		return new JsonResponse(service.getCurrentSession(SecurityContextHolder.getContext().getAuthentication(),
```
You may create static method in service that will hide work with SecurityContextHolder

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/controller/system/ExceptionHandlerControllerAdvice.java:
```diff
@@ -36,6 +40,29 @@
+	@ExceptionHandler(value = MethodArgumentNotValidException.class)
+	public JsonResponse methodArgumentNotValidExceptionHandler(MethodArgumentNotValidException e) {
+		logger.error("MethodArgumentNotValidException: {}", e.getMessage());
+		return new JsonResponse(Lists.newArrayList(Sets.newHashSet(e.getBindingResult().getFieldErrors().stream()
```
TODO

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/dto/RegisteredUserDTO.java:
```diff
@@ -0,0 +1,39 @@
+
+	@Override
+	public String toString() {
+		return "RegisteredUserDTO{" +
```
Look at ToStringBuilder

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/exception/system/Entity.java:
```diff
@@ -0,0 +1,35 @@
+	private static final Map<Entity, JsonError.Error> entityErrorMap;
+
+	static {
+		entityErrorMap = new ImmutableMap.Builder<Entity, JsonError.Error>()
```
ConcurrentHashMap

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/mail/PasswordRecoveryEmailMessage.java:
```diff
@@ -0,0 +1,75 @@
+
+	@Override
+	public String getPreparedEmailContent() {
+		return "<table border=\"0\" cellspacing=\"0\" width=\"100%\"><tr><td></td><td width=\"700\">" +
```
Any html build lib ?

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/security/AuthSuccessHandler.java:
```diff
@@ -57,40 +70,21 @@ protected void handle(
+			    authService.getCurrentSession(SecurityContextHolder.getContext().getAuthentication(),
+					    authTrustResolver)
+	    ));
+	    response.setStatus(HttpServletResponse.SC_OK);
```
Isn't it by default ?

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/security/AuthorizedUser.java:
```diff
@@ -77,10 +78,24 @@ public String toString() {
 
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
```
Equals and HashCode builders

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/security/CitizenAccessDeniedHandler.java:
```diff
@@ -0,0 +1,45 @@
+	) throws IOException, ServletException {
+		Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+
+		if (auth != null) {
```
auth instanceof AnonymousAuthenticationToken

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/security/LogoutSuccessHandler.java:
```diff
@@ -20,15 +22,26 @@
 			HttpServletResponse httpServletResponse,
 			Authentication authentication) throws IOException, ServletException {
-		new ObjectMapper().writeValue(httpServletResponse.getWriter(), authRestController.checkCurrSession(null));
+		new ObjectMapper().writeValue(
```
static constant

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/security/LogoutSuccessHandler.java:
```diff
@@ -20,15 +22,26 @@
+				authService.getCurrentSession(
+						SecurityContextHolder.getContext().getAuthentication(),
+						authTrustResolver));
 		httpServletResponse.setStatus(200);
```
Constant

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/service/AuthServiceImpl.java:
```diff
@@ -12,87 +12,144 @@
+				.setToken(DigestUtils.md5DigestAsHex((login + ip + LocalDateTime.now().toString()).getBytes()))
+				.build();
+
+		System.out.println("TOKEN: " + token);
```
log

yurakiziuk commented on src/main/java/com/shrralis/ssdemo1/service/AuthServiceImpl.java:
```diff
@@ -12,87 +12,144 @@
+		final User user = repository.getByLogin(dto.getLogin());
+
+		if (token == null) {
+			// TODO: throw new EntityNotFoundException(RECOVERY_TOKEN);
```
TODO

yurakiziuk commented on src/main/java/com/shrralis/tools/model/JsonError.java:
```diff
@@ -36,16 +44,34 @@ public String getErrmsg() {
+
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof JsonError) {
```
if (!obj instanceof ...
return false

yurakiziuk commented on src/main/java/com/shrralis/tools/model/JsonError.java:
```diff
@@ -84,14 +123,23 @@ public int getId() {
+				case BAD_FIELD_FORMAT:
+				case USER_ALREADY_EXISTS:
+					this.field = field;
+
```
remove

shrralis committed:
# Shrralis commit changelog #36:

##    [GLOBAL]:
     - Recoded with Roma's suggestions. (much thanks)

##    [Front-end]:
     `null`

##    [Back-end]:
     - Refactored some code.
     - New `AuthenticationFailureHandler`.

##    [Database]:
	 - Removed some unnecessary start data from change logs.

shrralis committed:
# Shrralis commit changelog #36:

##    [GLOBAL]:
     `null`

##    [Front-end]:
     `null`

##    [Back-end]:
     `null`

##    [Database]:
	 - Fixed `images.src` length to 128.

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/configuration/AppConfig.java:
```diff
@@ -29,9 +37,15 @@
+	private static final String EMAIL_PASS = "email.password";
+
+	@Resource
+	private Environment env;
```
@Value

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/exception/EntityNotExistException.java:
```diff
@@ -0,0 +1,66 @@
+		USER,
+		RECOVERY_TOKEN;
+
+		private static final Map<Entity, JsonError.Error> entityErrorMap;
```
upper case

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/security/CitizenAccessDeniedHandler.java:
```diff
@@ -0,0 +1,54 @@
+
+	private static final Logger logger = LoggerFactory.getLogger(CitizenAccessDeniedHandler.class);
+
+	private ObjectMapper mapper;
```
static final

Roma7-7-7 commented on src/main/java/com/shrralis/ssdemo1/security/CitizenAuthenticationFailureHandler.java:
```diff
@@ -0,0 +1,47 @@
+
+	@Autowired
+	public CitizenAuthenticationFailureHandler(ObjectMapper mapper) {
+		this.mapper = mapper;
```
No need to set it by constructor. static final constant

Roma7-7-7 reviewed (approved):
Object mapper should be removed from autowired constructor

shrralis committed:
# Shrralis commit changelog #36:

##    [GLOBAL]:
     - Last Roma's suggestions applied.

##    [Front-end]:
     `null`

##    [Back-end]:
     `null`

##    [Database]:
	 `null`

shrralis committed:
# Shrralis commit changelog #36:

##    [GLOBAL]:
     - Last Roma's suggestions applied.

##    [Front-end]:
     `null`

##    [Back-end]:
     `null`

##    [Database]:
	 `null`
